internal/jimm/juju/modelbuilder.go

```diff
@@ -227,6 +227,13 @@ func (b *modelBuilder) WithCloudCredential(credentialTag names.CloudCredentialTa
 	if b.err != nil {
 		return b
 	}
+
+	// Verify ownership of cloud credential
+	if b.owner == nil || b.owner.Name != credentialTag.Owner().Id() {
+		b.err = errors.E("invalid owner for cloud credential", errors.CodeUnauthorized)
```

Comment:
i would return `cloud credential not found` as the user does not have access to it and it gives away no information about cloud credentials.

Reply:
This also doesn't give information away I think. Because I haven't checked if the credential exists. I'm just validating the owner name of the credential (before we check if it exists).

Reply:
ha, i see.. but it might give somebody the impression that those cloud creds exist..

Reply:
also.. when would b.owner be nil? if that happens we're missing some information or have been running model builder steps in the wrong order?

Reply:
I don't think b.owner should be nil at this point, but if it is I didn't want to panic.

Reply:
>it might give somebody the impression that those cloud creds exist

Fair enough, will change.

Reply:
Went to change this and felt like "cloud credential not found" is also misleading, because the credential may very well exist. So I've changed it to "model owner doesn't match cloud-credential owner" which is clear, and doesn't leak information about what exists.